querybook/server/clients/github_client.py

```diff
@@ -0,0 +1,130 @@
+from flask import session as flask_session
+from github import Github, GithubException, Auth
+from typing import List, Dict, Optional
+
+from lib.github_integration.serializers import (
+    deserialize_datadoc_from_markdown,
+    serialize_datadoc_to_markdown,
+)
+from lib.logger import get_logger
+from models.datadoc import DataDoc
+from models.github import GitHubLink
+from env import QuerybookSettings
+
+LOG = get_logger(__name__)
+
+
+class GitHubClient:
+    def __init__(self, github_link: GitHubLink, datadoc: DataDoc):
+        """
+        Initialize the GitHub client with an access token from the session.
+        Raises an exception if the token is not found.
+        """
+        access_token = self._get_access_token()
```

Comment:
instead of getting the access token inside this class, how about passing it token as parameter so that this class can be decoupled from the `flask_session`?